portfolio/src/main/java/com/google/sps/servlets/TotalCommentsServlet.java

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.CommentsStore;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.MediaType;
+
+/** Servlet that returns the total number of comments. */
+@WebServlet("/total")
```

Comment:
For now it should be ok. However, if you crave for more, here are two ideas:
1. `/total` and `/data` are quite meaningless endpoints. Total of what? I know that for now there are only comments, but imagine adding more functionality in the future - it would be nice to have more specific url addresses, e.g. `/comments/total`.
2. Right now webpage need to send 2 requests on load: one to get comments, and the second to get total number of them. Each request comes with an overhead (network latency etc.) You might combine these 2 information together and return everything from /data, e.g such a JSON:
```
{
  comments: [...],
  total: X
}
```
On the other hand, this way servlet need to retrieve total number of comments every time someone fetches comments. SO there is some trade-off to consider.

Reply:
1. Totally agree.
2. I thought about it and I chose to separate the total because, besides the fact that I don't need the total every time, it wouldn't be immediate to include the total since the JSON string is created by the gson library directly from the objects.